include/mega/filesystem.h

```diff
@@ -22,7 +22,11 @@
 #ifndef MEGA_FILESYSTEM_H
 #define MEGA_FILESYSTEM_H 1
 
-#if defined (__linux__) || defined (__ANDROID__)
+#if defined (__linux__) && !defined (__ANDROID__)
+#include </usr/include/linux/magic.h>
+#endif
+
+#if defined (__linux__)
```

Comment:
Maybe it's good to remark, somewhere, that `__linux__` always include `__ANDROID__`??
```suggestion
#if defined (__linux__) || defined (__ANDROID__) // __ANDROID__ is always included in __linux__
```